play-json/jvm/src/main/scala/play/api/libs/json/JsonConfig.scala

```diff
@@ -186,9 +193,13 @@ object JsonConfig {
   private[json] def loadPreserveZeroDecimal: Boolean =
     prop(preserveZeroDecimalProperty, defaultPreserveZeroDecimal)(_.toBoolean)
 
-  // Default settings, which can be controlled with system properties.
-  // To override, call JacksonJson.setConfig()
-  val settings: JsonConfig =
+  /** Override configuration */
+  def setConfig(jsonConfig: JsonConfig): Unit = JacksonJson.initConfig(jsonConfig)
```

Comment:
There was concern in the original PR about thread-safety for exposing this publicly. However, I don't think that's really an issue. This config is only used to initialize `JacksonJson`, so any given operation will use a consistent set of classes.

Reply:
I don't think exposing such mutation in the API is a good idea